packages/server/lib/modes/run.js

```diff
@@ -1251,7 +1251,7 @@ module.exports = {
       //   await openProject.closeBrowser()
       // } else {
       debug('attempting to close the browser tab')
-      await openProject.closeBrowserTabs()
+      await openProject.resetBrowserTabsForNextTest(shouldKeepTabOpen)
```

Comment:
Passing this `shouldKeepTabOpen` variable around allows us to not open a new tab on the last spec. It's a lot of changes but there's probably a small performance improvement on the last spec at least.